src/pip/_internal/utils/appdirs.py

```diff
@@ -239,7 +239,9 @@ def _get_win_folder_with_ctypes(csidl_name):
         if ctypes.windll.kernel32.GetShortPathNameW(buf.value, buf2, 1024):
             buf = buf2
 
-    return buf.value
+    # On Python 2, ctypes.create_unicode_buffer().value returns "unicode",
+    # which isn't the same as str in the annotation above.
+    return buf.value  # type: ignore
```

Comment:
I would convert this to `str` before returning. There are already similar functions or lines of code in the code base, but I would first try converting using `sys.getfilesystemencoding()` and then fall back to other encodings if that doesn't work. I think it's important for this type annotation to be correct, otherwise it can lead to surprising `UnicodeError`s, etc. in the calling code since people looking at the types will be expecting `str` since that's what they'll see (and the type checker won't catch it either).

Reply:
I share the concern here -- however given my lack of experience w/ unicode issues and that this is Python 2 only, I personally *really* don't want to spend time working on it. 🙃

Plus it's in appdirs' internal copy so making those fixes should help too.

Reply:
Okay, then I would at least add a code comment near the function’s type annotation warning that it doesn’t return str under Python 2. Even if we don’t want to work on it, it’s still good to take steps to avoid introducing bugs when we’re touching the code.